frontend/benefit/applicant/src/components/applications/forms/application/step1/ApplicationFormStep1.tsx

```diff
@@ -0,0 +1,343 @@
+import { DynamicFormStepComponentProps } from 'benefit/applicant/types/common';
+import {
+  Checkbox,
+  Notification,
+  RadioButton,
+  SelectionGroup,
+  TextArea,
+  TextInput,
+} from 'hds-react';
+import React from 'react';
+import FormSection from 'shared/components/forms/section/FormSection';
+import { StyledFormGroup } from 'shared/components/forms/section/styled';
+
+import { APPLICATION_FIELDS } from '../../../constants';
+import {
+  StyledCompanyInfoColumn,
+  StyledCompanyInfoContainer,
+  StyledCompanyInfoRow,
+  StyledCompanyInfoSection,
+  StyledNotificationContent,
+  StyledSubSection,
+} from '../styled';
+import { useComponent } from './extended';
+
+const ApplicationFormStep1: React.FC<DynamicFormStepComponentProps> = ({
+  actions,
+}) => {
+  const {
+    t,
+    handleSubmit,
+    getErrorMessage,
+    fields,
+    translationsBase,
+    formik,
+  } = useComponent();
+
+  return (
+    <form onSubmit={handleSubmit} noValidate>
+      <FormSection header={t(`${translationsBase}.heading1`)}>
+        <StyledCompanyInfoContainer>
+          <StyledCompanyInfoSection>
+            <StyledCompanyInfoColumn>
+              <StyledCompanyInfoRow>Herkkulautanen Oy</StyledCompanyInfoRow>
+              <StyledCompanyInfoRow>Y-tunnus: 2114560-2</StyledCompanyInfoRow>
+            </StyledCompanyInfoColumn>
+            <StyledCompanyInfoColumn>
+              <StyledCompanyInfoRow>Keskuskatu 13 A 11</StyledCompanyInfoRow>
+              <StyledCompanyInfoRow>00100 Helsinki</StyledCompanyInfoRow>
+            </StyledCompanyInfoColumn>
+          </StyledCompanyInfoSection>
+          <StyledCompanyInfoSection>
+            <Notification
+              label={t(
+                `${translationsBase}.notifications.companyInformation.label`
+              )}
+              type="info"
+            >
+              <StyledNotificationContent>
+                {t(
+                  `${translationsBase}.notifications.companyInformation.content`
+                )}
+              </StyledNotificationContent>
+            </Notification>
+          </StyledCompanyInfoSection>
+        </StyledCompanyInfoContainer>
+        <StyledFormGroup>
+          <Checkbox
+            id={fields.hasCompanyOtherAddress.name}
+            name={fields.hasCompanyOtherAddress.name}
+            label={fields.hasCompanyOtherAddress.label}
+            required
+            checked={formik.values.hasCompanyOtherAddress === true}
+            errorText={getErrorMessage(
+              APPLICATION_FIELDS.HAS_COMPANY_OTHER_ADDRESS
+            )}
+            onChange={formik.handleChange}
+            onBlur={formik.handleBlur}
+            aria-invalid={
+              !!getErrorMessage(APPLICATION_FIELDS.HAS_COMPANY_OTHER_ADDRESS)
+            }
+          />
+        </StyledFormGroup>
+        {formik.values.hasCompanyOtherAddress && (
+          <StyledFormGroup>
+            <TextInput
```

Comment:
Some repetition in these TextInputs, perhaps they could be refactored per field. But as this is not final result, this can be done later.

Reply:
yes, there is repetition, i agree. but we are not going to create the form generator for one form. depending on field types there are different parameters, more generic you think, more overcomplication you get. Defining fields one by one, even with repetitive values, gives readability, compared to some config object you are looping through. if that was your concern.

Reply:
If you feel like that then it's better to keep that way 👍